pkg/deviceshifu/deviceshifuplc4x/deviceshifuplc4x.go

```diff
@@ -114,17 +114,22 @@ func (ds *DeviceShifu) writeCommandHandlerPlc4x() http.HandlerFunc {
 
 		plc4xRequest, err := request.Build()
 		if err != nil {
-			klog.Errorf("Error when build request by params, error: %v", err)
+			logger.Errorf("Error when build request by params, error: %v", err)
 			http.Error(w, "Error when build request with params, error: "+err.Error(), http.StatusBadRequest)
 			return
 		}
 
-		ctx, cancel := context.WithTimeout(context.Background(), time.Second*timeout)
-		defer cancel()
-		resultData := <-plc4xRequest.ExecuteWithContext(ctx)
+		var resultData model.PlcWriteRequestResult
+		select {
+		case resultData = <-plc4xRequest.Execute():
+		case <-time.After(time.Second * timeout):
+			logger.Errorf("Timeout when send request to device")
+			http.Error(w, "Timeout when send request to device", http.StatusInternalServerError)
+			return
```

Comment:
let's sync with main to avoid previous commits

Reply:
ok